### PR TITLE
support py3.6 and py3.9

### PR DESCRIPTION
**Pull request by graingert**

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -32,8 +32,12 @@ def get_long_description() -> str:
     package_dir={"": "src"},
     include_package_data=True,
     zip_safe=False,
-    install_requires=["anyio==1.*", "typing-extensions==3.7.*; python_version<'3.8'"],
-    python_requires=">=3.7",
+    install_requires=[
+        "anyio~=3.2",
```
Comment (florimondmanca) [inline setup.py:36]: Does this mean what I think it does, i.e. "3.2+, but below 4.0"?

**Review by florimondmanca — COMMENTED** [review]
Thanks! Lots of interesting things in here. I'm thinking of passing these in separate PRs just out of cleanliness and moving this forward without blocking on specific items.

File: setup.py (hunk 1 above)
Comment (graingert) [reply setup.py:36]: Yep

File: src/aiometer/_impl/amap.py [hunk 2]
```diff
@@ -16,10 +16,15 @@
 from .run_on_each import run_on_each
 from .types import T, U
 
-try:
-    from typing_extensions import Literal  # Python 3.7.
-except ImportError:  # pragma: no cover
-    from typing import Literal  # type: ignore
+if sys.version_info > (3, 8):  # pragma: no cover
```
Comment (graingert) [inline src/aiometer/_impl/amap.py:19]: it's better to use this approach than try/catch because mypy and pyupgrade both understand them

Comment (florimondmanca) [reply src/aiometer/_impl/amap.py:19]: I've seen this suggested before, sounds legit. :-)

**Review by florimondmanca — APPROVED** [review]
Fab, thanks. 3.6 support is cheap enough (only requiring `contextlib2` on 3.6) that I think we can be okay with adding support.

**Pull request merged**